docs/admin/federation/index.md

```diff
@@ -12,75 +13,132 @@ This guide explains how to set up cluster federation that lets us control multip
+[Previous Federation turn up mechanism](#previous-federation-turn-up-mechanism)
+at the end of this guide.
+
+### Initial setup
+
+Create a directory to store the configs required to turn up federation
+and export that directory path in the environment variable
+`FEDERATION_OUTPUT_ROOT`. This can be an existing directory, but it is
+highly recommended to create a separate directory so that it is easier
+to clean up later.
+
+```shell
+$ export FEDERATION_OUTPUT_ROOT="${PWD}/_output/federation"
+$ mkdir -p "${FEDERATION_OUTPUT_ROOT}"
+```
+
+Initialize the setup.
+
+```shell
+$ federation/deploy/deploy.sh init
+```
+
+Optionally, you can create/edit `${FEDERATION_OUTPUT_ROOT}/values.yaml` to
+customize any value in
+[federation/federation/manifests/federation/values.yaml](https://github.com/madhusudancs/kubernetes-anywhere/blob/federation/federation/manifests/federation/values.yaml). Example:
+
+```yaml
+apiserverRegistry: "gcr.io/myrepository"
+apiserverVersion: "v1.5.0-alpha.0.1010+892a6d7af59c0b"
+controllerManagerRegistry: "gcr.io/myrepository"
+controllerManagerVersion: "v1.5.0-alpha.0.1010+892a6d7af59c0b"
+```
+
+Assuming you have built and pushed the `hyperkube` image to the repository
+with the given tag in the example above.
 
 ### Getting images
 
-To run these as pods, we first need images for all the components. We can use
-official release images or we can build from HEAD.
+To run the federation control plane components as pods, you first need the
+images for all the components. You can either use the official release
+images or you can build them yourself from HEAD.
 
-#### Using official release images
+### Using official release images
 
-As part of every release, images are pushed to `gcr.io/google_containers`. To use
-these images, we set env var `FEDERATION_PUSH_REPO_BASE=gcr.io/google_containers`
-This will always use the latest image.
-To use the hyperkube image which includes federation-apiserver and
-federation-controller-manager from a specific release, we can set
-`FEDERATION_IMAGE_TAG`.
+As part of every Kubernetes release, official release images are pushed to
+`gcr.io/google_containers`. To use the images in this repository, you can
+set the container image fields in the following configs to point to the
+images in this repository. `gcr.io/google_containers/hyperkube` image
+includes the federation-apiserver and federation-controller-manager
+binaries, so you can point the corresponding configs for those components
+to the hyperkube image.
 
-#### Building and pushing images from HEAD
+### Building and pushing images from HEAD
 
-To run the code from HEAD, we need to build and push our own images.
-We can build the images using the following command:
+To run the code from HEAD, you need to build and push your own images.
+Check out the
+[Kubernetes repository](https://github.com/kubernetes/kubernetes)
+and run the following commands from the root of the source directory.
```

Comment:
The last two sentences are redundant. How about: 

"To build the binaries, check out the Kubernetes repository and run the following commands from the root of the source directory:"